dispatcher/backend/maint-scripts/report_youtube_api_keys.py

```diff
@@ -0,0 +1,126 @@
+#!/usr/bin/env python3
+
+"""Create a report of API keys usage in youtube recipes
+
+    ./report_youtube_api_keys.py
+
+    Configuration file named "report_youtube_api_keys.conf.json" must be placed in
+    same folder as script and contain a dictionnary of sha256(api_key) =>
+    api_key_display_name
+"""
+
+import datetime
+import hashlib
+import json
+import os
+import pathlib
+
+import requests
+import sqlalchemy as sa
+import sqlalchemy.orm as so
+from jinja2 import Environment, FileSystemLoader
+
+import db.models as dbm
+from db import dbsession
+
+
+@dbsession
+def report_youtube_api_keys(session: so.Session, *, display_unknown_secrets=False):
+    jinja_env = Environment(loader=FileSystemLoader("./"), autoescape=True)
+    jinja_template = jinja_env.get_template("report_youtube_api_keys.txt")
+    github_repo = os.environ["GITHUB_REPO"]
+    github_token = os.environ["GITHUB_TOKEN"]
+    github_issue_assignees = os.environ.get("GITHUB_ISSUE_ASSIGNEES", "").split(",")
+    github_issue_labels = os.environ.get("GITHUB_ISSUE_LABELS", "").split(",")
+    create_issue = os.environ.get("CREATE_ISSUE", "false").lower() == "true"
+
+    known_api_keys = json.loads(
+        pathlib.Path("report_youtube_api_keys.conf.json").read_text()
+    )
+    print("Listing schedules")
+    stmt = (
+        sa.select(dbm.Schedule)
+        .where(dbm.Schedule.config["flags"]["api-key"].astext.is_not(None))
```

Comment:
This is sqlAlchemy syntax. I imagine it doesn't fail if the flags dict doesn't have an `api-key` entry?

Why not selecting youtube schedules first to reduce the query scope?